### ISSUE TITLE
Frozen Baby Zombies

### ISSUE THREAD

**Issue by kandivia** [capnkirok/animaniamod#109]
### Unexpected Behavior
Baby Zombies have a chance when spawning into the world to become "frozen".
Affected Baby Zombies have the following traits:
-Have no functioning AI
-Can't be pushed/moved by the player (pistons can push them though)
-Can be hit once before being unable to be hit ever again
-Still counts to the mob count (affects mob spawner spawning)
-Disappears on world reload 

### Steps to Reproduce
Wait till a baby zombie that is affected by the bug spawns, (multiple zombie spawners in a test world helps)

### Version of Minecraft, version of Animania, Single Player or Server
MC 1.12.2
animania-1.12.2-1.5.1.1
SP & Server

### Screenshots encouraged
https://imgur.com/a/Ljz3y4Q

**Comment by Deshiba** [comment]
Animania doesn't mess with hostile mobs as far as I know. Do you have any other mods in use?

**Comment by kandivia** [comment]
Tested it with only:
animania-1.12.2-1.5.1.1
CraftStudio-1.0.0.93-mc1.12-alpha
on Forge 14.23.4.2705
I found it while playing a private modpack of my own, but trimmed the issue to this with testing.

**Comment by capnkirok** [comment]
Thanks for the report. There is a 'mob-riding' option in the config, which allows for baby zombies to ride animania chickens. My guess is the chickens are de-spawning and the zombie AI is confused (or only working server side). Will take a closer look.

In the meantime, you can turn this option off in the config (allowMobRiding = false)

**Comment by Tschipp** [comment]
Is this still an issue?

**Comment by kandivia** [comment]
I still got that test world, let me load the latest build onto it.

**Comment by kandivia** [comment]
Sure enough, spotted one after several minutes.
https://i.imgur.com/coYJGcO.png
I'll test the 'mob-riding' config again as well, but didn't seem to fix it last time I fiddled with it.
Edit: Well can confirm 'mob-riding' doesn't fix it. Caught it still spawning a chicken alongside the frozen zombie. 
https://i.imgur.com/GoSNwC8.png

**Comment by SiggiVG** [comment]
I've encountered this as well. And there's always a chicken wandering around near the frozen baby zombie. Will try disabling mob-riding in config and see if that relieves the issue.

**Comment by kandivia** [comment]
Small update: It would seem that this produces some log output. 
`[main/WARN] [minecraft/NetHandlerPlayClient]: Received passengers for unknown entity`
https://pastebin.com/fPrMcDKm
This line ouputs for every frozen zombie instance, which can spam the log heavily.
At first it seemed to output randomly, but after some testing it would seem to output very time the player moves back towards the frozen baby zombie at a point about 6 chunks between the player and the frozen zombie.

**Comment by Tschipp** [comment]
I still have never been able to reproduce this... I've found several baby zombies and they were all fine. My guess is that this was fixed when we decided to not replace vanilla animals with animania ones. The old chicken jockey of the zombie was probably removed by our code and replaced by a chicken (that the zombie was not riding however), which must have caused issue with its AI. I will close this in a week or so if I don't hear anything else.
